### PR TITLE
[vcpkg script] build_nmake use z_vcpkg_get_cmake_vars

### LINKED ISSUE #25840: vcpkg_build_nmake should call vcpkg_get_cmake_vars
currently it simply includes the toolchain file which is wrong and doesn't allow all cmake syntax to be used in it.

### PR DESCRIPTION
**Pull request by Neumann-A**
closes #25840

### CODE REVIEW

File: scripts/cmake/vcpkg_build_nmake.cmake [hunk 1]
```diff
@@ -70,7 +69,8 @@ function(vcpkg_build_nmake)
                     vcpkg_list(APPEND make_opts "INSTALLDIR=${install_dir_native}\\debug")
                 endif()
                 vcpkg_list(APPEND make_opts ${arg_OPTIONS} ${arg_OPTIONS_DEBUG})
-                set(ENV{CL} "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_DEBUG}")
+                set(ENV{CL} "$(VCPKG_DETECTED_CMAKE_CXX_FLAGS_DEBUG}")
```
Comment (Osyotr) [inline scripts/cmake/vcpkg_build_nmake.cmake:72]:
```suggestion
                set(ENV{CL} "${VCPKG_DETECTED_CMAKE_CXX_FLAGS_DEBUG}")
```

**Comment by JackBoosY** [comment]
pdcurses:
```
LINK : fatal error LNK1104: cannot open file 'link.exe'
```

detours:
```
detours.cpp(1035): error C3861: 'VirtualProtect': identifier not found
detours.cpp(1049): error C3861: 'VirtualProtect': identifier not found
detours.cpp(1105): error C3861: 'VirtualAlloc': identifier not found
detours.cpp(1152): error C3861: 'VirtualAlloc': identifier not found
detours.cpp(1422): error C3861: 'VirtualProtect': identifier not found
detours.cpp(1679): error C3861: 'SetThreadContext': identifier not found
detours.cpp(1695): error C3861: 'SetThreadContext': identifier not found
detours.cpp(1708): error C3861: 'VirtualProtect': identifier not found
detours.cpp(2116): error C3861: 'VirtualProtect': identifier not found
detours.cpp(2274): error C3861: 'VirtualProtect': identifier not found
detours.cpp(2359): error C3861: 'VirtualQueryEx': identifier not found
```

librttopo:
```
cl : Command line error D8048 : cannot compile C file 'src\box2d.c' with the /ZW option
```

**Comment by Neumann-A** [comment]
@JackBoosY: For `detours` and `librttopo` this probably means they don't support UWP? `pdcurses` uses `LINK` as an (env) variable which now gets sets and screws up the build (could use `_LINK_` however).

**Comment by dg0yt** [comment]
> librttopo:
> 
> ```
> cl : Command line error D8048 : cannot compile C file 'src\box2d.c' with the /ZW option
> ```

Reading the documentation of `/ZW`, I wonder why the `/ZW` should be used at all, for a C file that doesn't want "to support Microsoft C++ component extensions C++/CX"?

**Comment by Neumann-A** [comment]
> Reading the documentation of /ZW, I wonder why the /ZW should be used at all, for a C file that doesn't want "to support Microsoft C++ component extensions C++/CX"?

VS passes `/ZW` in the normal UWP App template. CMake also adds `/ZW:nostdlib` for store apps.

**Comment by JackBoosY** [comment]
Any news?

**Comment by Neumann-A** [comment]
this was green in https://github.com/microsoft/vcpkg/pull/25862/commits/b28cb4c753a99c54876116f5e0ca39ecb6b5e977 so just fix the merge conflicts. I am reluctant to do it. The PR has been green for 1 month and the other detours PR was a lot younger. (like opened only two days ago)

**Review by JackBoosY — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thanks!

**Comment by dg0yt** [comment]
> > > librttopo:
> > > ```
> > > cl : Command line error D8048 : cannot compile C file 'src\box2d.c' with the /ZW option
> > > ```
> > 
> > Reading the documentation of `/ZW`, I wonder why the `/ZW` should be used at all, for a C file that doesn't want "to support Microsoft C++ component extensions C++/CX"?
> 
> VS passes `/ZW` in the normal UWP App template. CMake also adds `/ZW:nostdlib` for store apps.

I think I hit the right spot: `/ZW:nostdlib` comes from pushing CXX flags into `_CL_` , so it breaks any C project for uwp:
librttopo (this PR) 
readosm (#26455) 
openssl (#27150)

Cf. `cmake-get-vars-x64-uwp-dbg.cmake.log`:
~~~cmake
set(VCPKG_COMBINED_CXX_FLAGS_DEBUG "-DWIN32 -D_WINDOWS -D_UNICODE -DUNICODE -DWINAPI_FAMILY=WINAPI_FAMILY_APP -D__WRL_NO_DEFAULT_LIB__ -nologo -Z7 -MP -GS -Gd -Gm- -W3 -WX- -Zc:wchar_t -Zc:inline -Zc:forScope -fp:precise -Oy- -EHsc  -utf-8  -D_DEBUG -Od -RTC1 -MDd   -ZW:nostdlib")
set(VCPKG_COMBINED_C_FLAGS_DEBUG   "-DWIN32 -D_WINDOWS -D_UNICODE -DUNICODE -DWINAPI_FAMILY=WINAPI_FAMILY_APP -D__WRL_NO_DEFAULT_LIB__ -nologo -Z7 -MP -GS -Gd -Gm- -W3 -WX- -Zc:wchar_t -Zc:inline -Zc:forScope -fp:precise -Oy- -EHsc  -utf-8  -D_DEBUG -Od -RTC1 -MDd  ")
~~~

**Comment by Neumann-A** [comment]
So we need a `LANGUAGE` argument for `vcpkg_configure_nmake` and switch the flag accordingly?

**Comment by dg0yt** [comment]
The language argument is the primary solution. I only wonder if we want to do more work in the global script, or if it should go to a helper port.

**Comment by Neumann-A** [comment]
> I only wonder if we want to do more work in the global script, or if it should go to a helper port.

What additional steps are you thinking of?

**Comment by dg0yt** [comment]
Option to use jom (and maybe filter out /MP then).
Option to not set `_CPP_` at all (and use OPTIONS_DEBUG/RELEASE instead).
Pass multiple targets.
Pass the name of the project's install dir variable.

**Comment by Neumann-A** [comment]
I completely misunderstood. I thought you meant new helper ports with different functionality. But you only meant to move the stuff here to a helper port called `vcpkg-nmake`.

**Pull request merged**